url10fb6bd43e_guwere_P2PSearch_tgz-pJ8-MainJ8/src/Receiver.java

```diff
@@ -0,0 +1,38 @@
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.util.concurrent.BlockingQueue;
+
+public class Receiver implements Runnable {
+    private DatagramSocket socket;
+    private Controller controller;
+    private BlockingQueue receiveQueue;
+
+    public Receiver(DatagramSocket socket,BlockingQueue rQueue){
+        this.socket = socket;
+        this.controller = controller;
+        receiveQueue = rQueue;
+    }
+
+    public void run(){
+        while(true){
+            byte[] buf = new byte[NetworkProperties.MTU];
+            DatagramPacket message = new DatagramPacket(buf, buf.length);
+            try {
+                System.out.println("Waiting for message...");
+                socket.receive(message);
+                receiveQueue.put(message);
+
+            } catch (IOException e) {
+                System.err.println("Error receiving packet in Receiver");
+                e.printStackTrace();
+            } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)